src/main/java/com/watch/shop/app/controller/WatchController.java

```diff
@@ -0,0 +1,54 @@
+package com.watch.shop.app.controller;
+
+import com.watch.shop.app.model.entity.Watch;
+import com.watch.shop.app.model.service.WatchService;
+import com.watch.shop.app.view.WatchView;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public class WatchController {
+
+    private final WatchService watchService;
+    private final WatchView watchView;
+    private final MenuHandler menuHandler;
+
+    public void run() {
+        boolean running = true;
+
+        while (running) {
+            String choice = menuHandler.printMainMenu();
+
+            switch (choice) {
+                case "1" -> menuHandler.displayAllWatches(watchService.getWatches());
+                case "2" -> sortChoice();
+                case "3" -> menuHandler.displayTotalCost(watchService.getTotalCost());
+                case "4" -> {
+                    try {
+                        Watch watch = menuHandler.printNewWatchMenu();
+                        watchService.insertWatch(watch);
+                    } catch (Exception e) {
+                        watchView.printMessage("\nInvalid input");
+                    }
+                }
+                case "5" -> running = false;
+                default -> watchView.printMessage("\nInvalid choice");
+            }
+        }
+    }
+
+    private void sortChoice() {
+        boolean running = true;
+
+        while (running) {
+            String param = menuHandler.printSortMenu();
+
+            switch (param) {
+                case "1" -> menuHandler.displayAllWatches(watchService.sortWatchesByParam("price"));
```

Comment:
Don't use plain strings, one changes or typo and code will stop work.

Use or enum, or shared constants, or better, just put in service three method
sortWatchByPrice()
..ByColor()
ByArrivalDate

And in service just use 1 common private method

.getWatchesSortedByPrice() {
   getWatchesSortedByParam( some comparator);
   }
   ..
   private List<Watch> getWachesSortedByParam(Comparator<Watch> comparator)) {
     return...
     }
     
     so you will have three method for controller, and all they will call 1 shared private method 
     
     Try to avoid keys with plain string, you could accidently just remove 1 symbol and logic will crash.